### PR TITLE
refactor(http): remove https from server

### LINKED ISSUE #455: Serve Node over HTTP, use nginx for HTTPS.
_This issue is imported from @sfount's 1.X issue found [here](https://github.com/IMA-WorldHealth/bhima-1.X/issues/707)._

Currently our node server uses the HTTPS protocol to serve BHIMA. I have found that this can be inconvenient whilst developing, it also creates large warnings about how the website is 'untrusted'. 

This is not a problem if the server is configured correctly with the correct SSL certificates installed, however I worry that this is a complicated process for someone who does not fully understand networking. 

My suggestion (for this discussion) is to use a **reverse proxy** to encrypt the network traffic and run our node server through HTTP. 

This would look a bit like this 

`Client` <-> `Reverse proxy (HTTPS server)` <-> `Node BHIMA server (HTTP)`

A very popular and widely used reverse proxy server is called [nginx](http://wiki.nginx.org/Main). We can configure this with one configuration file (nginx.conf) to act as a reverse proxy for the BHIMA server. Here are some pros and cons to using a separate process for HTTPS:

Pros 
- Very quick for development - simply type localhost, no worrying about certificates or trusting any sites
- Requires no additional network experience for hospital installs - if the hospital does not have the experience to set up HTTPS they do not need to and will receive no warnings about the untrusted website 
- If certificates expire or the configuration is broken the Node server can still be accessed through HTTP 
- Reverse proxy allows load balancing across many cores and machines if BHIMA is ever big enough 
- Purely optional - if the hospital is equipped to encrypt their data connection they can add this nginx instance
- Clear separation of networking jobs/ development and BHIMA application development 

Cons 
- It is no longer mandatory to serve the connection over HTTPS - HTTP is **very** vulnerable to hacking 
- If the nginx server breaks (for some reason) that is an additional thing to debug 

This is just a suggestion/ discussion - we can decide on if this is right for BHIMA or not.

### PR DESCRIPTION
**Pull request by jniles**
**BREAKING CHANGE**

This commit removes HTTPS support from bhi.ma.  HTTPS is best handled by a reverse proxy.  See [this article](https://expressjs.com/en/guide/behind-proxies.html) for more details on running the server behind a proxy.

Because of this change, there is no more need to store self-signed certificates.  They have been removed.

To update your code you should:
1. Remove TLS_\* variables from your .env.\* files.
2. Remove HTTPS=1 from your .env.\* files.
3. Clear the browser cache.

Closes #455.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by DedrickEnc** [comment]
I am reading the article

File: server/config/express.js [hunk 1]
```diff
@@ -43,10 +43,16 @@ exports.configure = function configure(app) {
     resave: Boolean(process.env.SESS_RESAVE),
     saveUninitialized: Boolean(process.env.SESS_UNINITIALIZED),
     unset: process.env.SESS_UNSET,
-    cookie: { secure : true },
+    cookie: { httpOnly : true },
     retries: 50
   }));
 
+  // indicate that we are running behind a trust proxy and should use a secure cookie
```
Comment (DedrickEnc) [inline server/config/express.js:50]: What will happen if we are in a development environnement

Comment (DedrickEnc) [inline server/config/express.js:50]: According to the article the sub net or some address should be set even if it is not important so what will be our position we will configure the sub net or not? if yes how? (Notice that we will face many type of network in terms of address configuration)

**Review by DedrickEnc — COMMENTED** [review]
@jniles 
I expressed some concerns

File: server/config/express.js (hunk 1 above)
Comment (jniles) [reply server/config/express.js:50]: There address is local.  There is no configuration necessary, the only thing we need to do is set `trust proxy` to `true` in production and listen on a high number port (like 3000).

We can then put any proxy (Apache, Nginx, HAproxy, etc) on port 80 and 443 of the same machine and route all traffic to localhost:3000 (where the express server is).

For a boolean value, nothing else needs to be configured.

Comment (jniles) [reply server/config/express.js:50]: Nothing.  The application will not do any extra checks on the cookie for tampering or check HTTP headers.

**Comment by mbayopanda** [comment]
@jniles everything work very well in http mode 👍 

I know that it is not your codes but i noticed that the e2e test run only `test/end-to-end/user/user.spec.js` 

user spec

**Review by mbayopanda — APPROVED** [review]
The migration from https to http is so clear 👍 and it works very well.

**Comment by jniles** [comment]
@mbayopanda, thanks.  I've made the change.  Things pass locally, but we'll see if Travis breaks.

**Pull request merged**